backend/src/packages/auth/auth.service.ts

```diff
@@ -87,6 +106,61 @@ class AuthService {
     return { ...userWithToken, group };
   }
 
+  public async signUpBusiness(
+    payload: BusinessSignUpRequestDto,
+  ): Promise<UserEntityObjectWithGroupAndBusinessT> {
+    const isExistingUser = await this.checkIsExistingUser(payload);
+
+    const {
+      phone,
+      email,
+      firstName,
+      lastName,
+      password,
+      companyName,
+      taxNumber,
+    } = payload;
+
+    if (isExistingUser) {
+      throw new HttpError({
+        message: HttpMessage.USER_EXISTS,
+        status: HttpCode.CONFLICT,
+      });
+    }
+
+    const group = await this.groupService.findByKey(UserGroupKey.BUSINESS);
+
+    if (!group) {
+      throw new HttpError({
+        message: HttpMessage.INVALID_GROUP,
+        status: HttpCode.BAD_REQUEST,
+      });
+    }
+
+    const newUser = await this.userService.create({
+      phone,
+      email,
+      firstName,
+      lastName,
+      password,
+      groupId: group.id,
+    });
+
+    const userWithToken = await this.generateAccessTokenAndUpdateUser(
+      newUser.id,
+    );
```

Comment:
can we create access token and then create user to reduce db calls amount?

Reply:
We need to know a userId for token. So we need do it in two steps: create a user and receive an id then create a token and update the user.

Reply:
Right, my bad, thanks!